### PR TITLE
Optimise fast transfer message change

### PR DESCRIPTION
**Pull request by sixstone-qq**
By doing two changes:

* Perform the search for message identifiers along with the change number to discard a change whose is already included in the `cnset_seen` before opening the message and get all its available properties.
* Set sort order in changed messages before getting the table row count and conditionally

Use case:
No infinite loop with no full synced folder in Outlook 2013 when you were in the middle of the initial synchronisation in a business size folder.

### CODE REVIEW

**Comment by jkerihuel** [comment]
From code perspective, minor questions asked, LGTM to me. :+1:
Nevertheless need independent functional testing from any other reviewer.

File: mapiproxy/servers/default/emsmdb/oxcfxics.c [hunk 1]
```diff
@@ -790,15 +791,17 @@ static void oxcfxics_table_set_cn_restriction(struct emsmdbp_context *emsmdbp_ct
...
 	struct emsmdbp_object		*table_object, *message_object;
 	uint32_t			i;
-	static enum MAPITAGS		mid_property = PidTagMid;
 	enum MAPISTATUS			*retvals, *header_retvals, retval_msg_class = MAPI_E_NOT_FOUND;
+	const enum MAPITAGS		table_prop_tags[] = {PidTagMid, PidTagChangeNumber};
 	void				**data_pointers, **header_data_pointers;
 	struct FILETIME			*lm_time;
 	NTTIME				nt_time;
 	uint32_t			unix_time, contextID, message_class_prop_index;
+	const uint32_t			TABLE_PROPS_COUNT = sizeof(table_prop_tags) / sizeof(enum MAPITAGS);
```
Comment (blaxter) [inline mapiproxy/servers/default/emsmdb/oxcfxics.c]: I agree that this should be downcase (I know there are functions with variables like this, upcase, but I never liked it).

**Comment by blaxter** [comment]
:+1: waiting for the 2 minor changes to merge

**Comment by sixstone-qq** [comment]
Everything updated upon request. Thanks very much for your feedback! Really appreciated.

**Comment by blaxter** [comment]
No problem darling

**Comment by exekias** [comment]
jenkins retest this please

**Pull request merged**